### DISCUSSION TITLE
Using Contentment in a class library project in .NET 6

### DISCUSSION THREAD

**Discussion by rosspetev** [leekelleher/umbraco-contentment#257] [Question] [answered]
I have a two - project Umbraco solution; a core project + a web project. 
I have the Contentment dependency in the Core project since I have an implementation of a **IDataSource** .

After upgrading to **.NET 6**, I get "**Found multiple publish output files with the same relative path**" error when publishing the app. 
I think this is due to the app_plugins files being copied so they are present in both projects.
I suspect this is similar but can't be sure: https://github.com/leekelleher/umbraco-contentment/discussions/237

Would be grateful for solution to this or a workaround (I guess one would be to move any custom IDataSource to the web project).

Thank you for your time and efforts!

**Comment by leekelleher** [comment, marked as answer]
Hi @rosspetev, which of the NuGet packages are you referencing in your class library project?

It's better to use [`Our.Umbraco.Community.Contentment.Core`](https://www.nuget.org/packages/Our.Umbraco.Community.Contentment.Core/) NuGet package, (which only contains the assembly DLL) instead of the main [`Our.Umbraco.Community.Contentment`](https://www.nuget.org/packages/Our.Umbraco.Community.Contentment) NuGet package, (which contains the `App_Plugins/Contentment` folder).

**Reply by rosspetev** [reply]
Thanks Lee, that works. Don't know why I didn't see the .Core package.

**Reply by leekelleher** [reply]
No problem. I'm wondering if I'm missing anything with the docs/README? Let me know if there's anything that might have helped. Thanks. 👍

**Reply by rosspetev** [reply]
No, I think it's all very well described! I guess I had the right thought was to look for a ".Core" package but must have looked in the "Installed packages" tab..
